### DISCUSSION TITLE
Multicast networking?

### DISCUSSION THREAD

**Discussion by pgcudahy** [abiosoft/colima#462] [Q&A]
Hello, just started using colima which has been a really nice experience but have a question. [Home-assistant](https://www.home-assistant.io/) is a home automation service that relies on multicast to auto-discover network services. They [recommend](https://www.home-assistant.io/installation/linux#install-home-assistant-container) using the `network_mode: "host"` option in docker to achieve this but that's not possible on VM based docker for mac. Is there any way to get around this?

**Comment by abiosoft** [comment]
Technically, you can set the network mode to `host` but it is only host to the VM, not the macOS host.

**Reply by pgcudahy** [reply]
Right, but the VM won't see network multicast packets. Is there a way that those can be bridged from the macOS host to the VM?

**Reply by abiosoft** [reply]
[socket_vmnet](https://github.com/lima-vm/socket_vmnet) actually supports both bridge and host networks as well. 
However, I did not explore those as they are not the use cases for the average Colima user.

Sounds like you need bridge mode, which should make the VM visible to other computers on your network.

**Reply by pgcudahy** [reply]
Updated 2022-11-06
Now with this integrated in the github version it's even easier
```
brew install --HEAD colima
brew install docker docker-compose
mkdir -p ~/.docker/cli-plugins
ln -sfn $(brew --prefix)/opt/docker-compose/bin/docker-compose ~/.docker/cli-plugins/docker-compose
colima start --network-address
docker-compose -f docker-compose.yml up -d
```
With the standard docker-compose settings from [home-assistant](https://www.home-assistant.io/installation/linux#docker-compose)

I tested pinging my mdns printer from both `colima ssh` and `docker exec -it homeassistant /bin/bash` and it works in both!

**Reply by SteveMarshall** [reply]
If I'm reading this right, this means your VM/containers can see mdns devices elsewhere on the network, but what about the containers being addressable from other devices on the network?

I've a couple of things I run that need to be exposed as separate devices on the network, preferably with their own IPs and so forth. Is that what you've got here, @pgcudahy, or does your home-assistant present itself as being a service on your Mac?

**Reply by abiosoft** [reply]
Colima uses shared network and therefore the IP address is not reachable by other devices on the network.

**Reply by eelcocramer** [reply]
> Updated 2022-11-06 Now with this integrated in the github version it's even easier
> 
> ```
> brew install --HEAD colima
> brew install docker docker-compose
> mkdir -p ~/.docker/cli-plugins
> ln -sfn $(brew --prefix)/opt/docker-compose/bin/docker-compose ~/.docker/cli-plugins/docker-compose
> colima start --network-address
> docker-compose -f docker-compose.yml up -d
> ```
> 
> With the standard docker-compose settings from [home-assistant](https://www.home-assistant.io/installation/linux#docker-compose)
> 
> I tested pinging my mdns printer from both `colima ssh` and `docker exec -it homeassistant /bin/bash` and it works in both!

Your update suggests that multicast traffic from the network should be received by the `colima` VM guest when using the `--network-address` switch. When I try this I do get prompted for a password when `colima` starts and `colima ls` output displays an IP address but I'm not able to resolve bonjour services from within a docker container that uses `host` networking. Should this work out-of-the-box or does this require a different network driver?
